doc/type_requirements/File.qbk

```diff
@@ -0,0 +1,184 @@
+        lowest byte offset for which a read will always return a
+        [link nudb.ref.nudb__error short_read] error. Behavior is undefined
+        if `f` does not refer to an open file.
+    ]
+]
+[
+    [`f.read(o,a,n,ec)`]
+    [`void`]
+    [
+        Attempts to read `n` bytes starting at file offset `o`
+        from the open file referred to by `f`. Bytes read are stored
+        in the memory buffer at address 'a'. The memory buffer at 'a'
+        must be at least n bytes long. If an error occurs, `ec` is set
+        to the system specific error code. Behavior is undefined if `f`
+        does not refer file opened with [link nudb.ref.nudb__file_mode file_mode]
+        equal to scan or read.
+    ]
+]
+[
+    [`f.sync(ec)`]
+    [`void`]
+    [
+        Attempts to synchronize the file on disk. This instructs the
+        operating system to ensure that any data which resides in caches
+        or buffers is fully written to the underlying storage device
+        before this call returns. If an error occurs, `ec` is set to the
+        system specific error code. Behavior is undefined if
+        `f` does not refer to an open file.
+
+        NuDB's database integrity guarantees are only valid if the
+        implementation of `sync` assures that all data is fully written
+        to the underlying file before the call returns.
+    ]
+]
+[
+    [`f.write(o,a,n,ec)`]
+    [`void`]
+    [
+        Attempts to write `n` bytes starting at file offset `o`
+        to the open file referred to by `f`.  The memory buffer at
+        `a` should contain the `n` bytes meant to be copied to the file.
+        If an error occurs, `ec` is set to the system specific error code.
+        Behavior is undefined if `f` does not refer a file opened with
+        [link nudb.ref.nudb__file_mode file_mode] equal to write or append.
+    ]
+]
+[
+    [`f.trunc(o,ec)`]
+    [`void`]
+    [
+        Attempts to change the size of the open file referred to by `f`
+        to the size in bytes specified by `o`. If an error occurs, `ec`
+        is set to the system specific error code. Behavior is undefined if
+        `f` does not refer to an open file. After a successful call,
+        `a.size(ec)` will return `o`.
+
+        NuDB's database integrity guarantees are only valid if the
+        implementation of `trunc` assures that subsequent calls to
+        `size` will return `o`, even if the program is terminated or the
+        device is taken offline before calling `size`.
+    ]
+]
+[
+    [`is_File<F>`]
+    [`std::true_type`]
+    [
+        returns `std::true_type` for `F`. Otherwise returns `std::false_type`]
+]
+]
+
+[heading Associated Functions]
+* [link nudb.ref.nudb__block_size block_size]
+* [link nudb.ref.nudb__erase_file erase_file]
+
+[heading Models]
+* [link nudb.ref.nudb__posix_file posix_file]
+* [link nudb.ref.nudb__win32_file win32_file]
+* [link nudb.ref.nudb__native_file native_file]
```

Comment:
The idea of "Models" is also extremely common in the documentation of templates.  It's very unfortunate that the originators of STL used very misleading names for their entities.  

Concept -> Type Requirements
Model -> Type Instance
Associated Type -> good no change
Associated Function -> I invented that here.  You have an unusually situation where a number of free functions are (open etc.) aren't member functions because the store is only an object when it's been open.  More common would store being a serializable type so that once created with the file names it would be stored as a container of 3 file names.  In this case open, close, rekey, etc would be member functions of store.  In this case "Associated Functions" wouldn't be necessary as they would be member functions.  In this case the functions only make sense in the context of a data file triplet so they are not really "free" functions and they are not member functions either so I made up my own term and grouped them all together.